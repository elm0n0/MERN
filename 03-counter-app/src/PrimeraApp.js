import React from 'react';

const PrimeraApp = ( {saludo = 'Hola Mundo'} ) => {


    return ( 
        <>
        <h1>{saludo}</h1>
        {/* <pre>{JSON.stringify(saludo,null,3)}</pre> */}
        <p>My Primera App</p>
        </>
    )
}

export default PrimeraApp;