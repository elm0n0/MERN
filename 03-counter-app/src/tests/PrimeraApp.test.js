import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';


describe('PrimeraApp', () => {
    test('debe mostrar el componente app correctamente ', () => {
        const saludo = "Hola, Soy Goku";
        const wraper = shallow(<PrimeraApp saludo = {saludo} />)
        expect(wraper).toMatchSnapshot()
    })
    test('debe de mostrar el subtitulo enviado por props ', () => {
        const saludo = "Hola, Soy Goku";
        const subtitulo = "Soy un subtitulo";
        const wraper = shallow(
            <PrimeraApp 
                saludo = {saludo} 
                subtitulo = {subtitulo}
            />)
        const textoParrafo = wraper.find('p').text();
        console.log(textoParrafo);
        expect(textoParrafo).toBe(subtitulo)
    })
    
})