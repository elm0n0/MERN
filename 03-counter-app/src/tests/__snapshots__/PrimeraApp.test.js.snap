// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeraApp debe mostrar el componente app correctamente  1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    Soy un Subtitulo
  </p>
</Fragment>
`;
